corehq/apps/reports/tasks.py

```diff
@@ -224,28 +223,42 @@ def send_email_report(self, recipient, request, content, subject, config):
     '''
     from corehq.apps.reports.views import render_full_report_notification
 
-    size_too_large = False
     body = render_full_report_notification(request, content).content
     try:
-        send_html_email_async(subject, recipient,
-                              body, email_from=settings.DEFAULT_FROM_EMAIL,
-                              smtp_exception_skip_list=[522])
+
+        send_HTML_email(subject, recipient,
+                        body, email_from=settings.DEFAULT_FROM_EMAIL,
+                        smtp_exception_skip_list=[522])
+
     except Exception as er:
         if getattr(er, 'smtp_code', None) == 522:
-            size_too_large = True
+            # If the smtp server rejects the email because of its large size.
+            # Then sends the report download link in the email.
+            email_large_report(request, config, recipient)
         else:
             self.retry(exc=er)
 
-    # If the smtp server rejects the email because of its large size.
-    # Then sends the report download link in the email.
-    if size_too_large:
-        report = config.report(request, domain=config.domain, **{})
-        report.rendered_as = 'export'
-        report.decorator_dispatcher(request, domain=config.domain,
-                                    report_slug=config.report_slug,
-                                    *(), **{})
-        export_all_rows_task.delay(report.__class__, report.__getstate__(),
-                                   recipient=recipient)
+
+def email_large_report(request, config, recipient):
+    """
+    Function sends the requested report download link in the email.
+    This function is invoked when user tries to email very large report.
+
+    :Parameter request:
+        request object which contains request details
+    :Parameter config:
+            object of ReportConfig. Contains the report configuration
+            like reportslug, report_type etc
+    :Parameter recipient:
+            recipient to whom email is to be sent
+    """
+    report = config.report(request, domain=config.domain, **{})
```

Comment:
what is this line and the next two for? `config.report` should already have the instantiated report object, so you should be able to use  `.__class__` and `.__getstate__()` directly on that

Reply:
is the empty dict necessary. I imagine all the other args must be optional if you can pass an empty one, and it adds a bit of clutter, but I could be misunderstanding what is going on here.